index.js

```diff
@@ -837,6 +837,8 @@ function findRefs (obj, options) {
  *
  * @returns {Promise} a promise that resolves a {@link module:JsonRefs~RetrievedRefsResults}
  *
+ * @throws {Error} if the provided `location` argument is not a string
```

Comment:
How about something like this:

```js
...
 * @returns {Promise} a promise that resolves a {@link module:JsonRefs~RetrievedRefsResults} and rejects with an `error` if input arguments fail validation or there is an issue retrieving your remote references *(when necessary)*
...
```

This means no confusing `@throws` tag.  I'd do the same for any API that returns a `Promise`.